### ISSUE TITLE
Errors in running Demo project

### ISSUE THREAD

**Issue by TheMadPope** [apexsharp/apexsharp#13]
Errors on running Demo project. Connected successfully, but the Upsert test fails.
Down in SoqlApi.ConvertSoql, it's passing these vars:
string soql = "SELECT Id, Email FROM Contact WHERE Id = :contactNew.Id"
params object[] param = 
![image](https://user-images.githubusercontent.com/13425302/34276610-098eefe2-e657-11e7-85aa-835e9ee3581c.png)

Which parses down to "SELECT Id, Email FROM Contact WHERE Id = 'Apex.System.ID'" -- Which is obviously not the Id. So the Soql query returns an unhandled exception.

Even stepping over that Upsert call, it appears to be making changes to ~33 classes in the CSharpClasses folder, adding a reference to "ApexSharpApi.ApexApi", which is invalid.

Going to comb through what else I can find. Unsure if I missed something? I'll update this if I can uncover my error.

**Comment by jayonsoftware** [comment]
Can you pull the dev branch, I have created an example of upsert. Please try it and let me know

**Comment by yallie** [comment]
I've just uploaded the new ApexParser nuget version with upsert support. I'll take about ten minutes to validate and publish, then I'll update the demo. -- Regards, Alexey   22.12.2017, 01:43, "Jay Janarthanan" <notifications@github.com>:Can you pull the dev branch, I have created an example of upsert. Please try it and let me know—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by yallie** [comment]
My mistake, sorry!
This issue has nothing to do with the upsert support in ApexParser.

**Comment by TheMadPope** [comment]
I'm going to keep digging. I'm having a number of problems getting this to dance for me. Excited about the project, however. I'm alright with closing this issue, I think it generally has to do with not performing out of the box, as it were. But it looks like there's a number of issues preventing that. Thanks!

**Comment by jayonsoftware** [comment]
I am assuming you are using Visual Studio on a PC ?

**Comment by TheMadPope** [comment]
Yes, VS2017 on PC. Had to manually create a json file or two, and has some anomalies on the second time running Demo. I'll meet with this matter and see if I can get consistent notes for you.

**Comment by jayonsoftware** [comment]
I have been working on a new setup on the dev branch, let me update the readme and push to the main branch. give a try in the morning.

**Comment by jayonsoftware** [comment]
I have made changes to make things easy. Please get the latest master branch and give it a try.
